narwhals/_pandas_like/utils.py

```diff
@@ -417,9 +417,16 @@ def reverse_translate_dtype(  # noqa: PLR0915
 
 
 def validate_indices(series: list[PandasLikeSeries]) -> list[Any]:
```

Comment:
both here, and in the new function for pyarrow, maybe let's call them `broadcast_series`?